qadence/ml_tools/train_grad.py

```diff
@@ -110,8 +110,11 @@ def loss_fn(model: torch.nn.Module, data: torch.Tensor) -> tuple[torch.Tensor, d
     """
     # load available checkpoint
     init_iter = 0
+    log_device = "cpu" if device is None else device
```

Comment:
@dominikandreasseitz in Qadence is this guaranteed?
Because it's not in "regular" PyTorch.

Reply:
What do you mean by guaranteed? I think this is needed because the default value for `device` argument in the `train_grad()` function is `None`, so with this if the user does not pass a device to `train_grad` it will just try to load a ckpt from `cpu`. Or do you mean there could a problem with the `if` statement itself?

Reply:
device of `None` in PyTorch means don't change it.
So you have a tensor on GPU, and move it to `None` it remains on the same GPU
```python
awennersteen:~ $ python3
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> a = torch.rand(5, device="cuda")
>>> a.to(None)
tensor([0.0810, 0.8175, 0.8455, 0.5090, 0.4081], device='cuda:0')
>>> 

```

Reply:
So in general, someone could setup their code in advance to work on a certain GPU, and e.g. set `torch.default_device` to `"cuda:0"` and pass `device=None` but have the data on GPU, not cpu.

I just don't know if that is true on Qadence since we do mess around a bit with the parameters and such, and I hope @dominikandreasseitz knows

Reply:
I think it is. All examples of setting the device are mainly in the [examples](https://github.com/search?q=repo%3Apasqal-io%2Fqadence%20default_device&type=code). But I would seek for @dominikandreasseitz on this.